src/core/bonded_interactions/angle_cosine.hpp

```diff
@@ -27,37 +27,65 @@
  */
 
 #include "angle_common.hpp"
-#include "bonded_interaction_data.hpp"
 
 #include <utils/Vector.hpp>
 #include <utils/math/sqr.hpp>
 
 #include <cmath>
 #include <tuple>
 
-/** Set parameters for the angle potential. */
-int angle_cosine_set_params(int bond_type, double bend, double phi0);
+/** Parameters for three-body angular potential (cosine). */
+struct AngleCosineBondParameters {
+  /** bending constant */
+  double bend;
+  /** equilibrium angle (default is 180 degrees) */
+  double phi0;
+  /** cosine of @p phi0 (internal parameter) */
+  double cos_phi0;
+  /** sine of @p phi0 (internal parameter) */
+  double sin_phi0;
+
+  double cutoff() const { return -1.; }
+
+  static constexpr int num = 2;
+
+  AngleCosineBondParameters() = default;
+  AngleCosineBondParameters(double bend, double phi0);
+
+  std::tuple<Utils::Vector3d, Utils::Vector3d, Utils::Vector3d>
+  angle_force(Utils::Vector3d const &r_mid, Utils::Vector3d const &r_left,
```

Comment:
i'm not sure about ```angle_force```... wouldn't ```forces``` more precisely reflect that there are multiple forces returned and calculated?